src/Page/Packages.elm

```diff
@@ -148,259 +190,357 @@ update navKey msg model =
 view : Model -> Html Msg
 view model =
     Search.view { toRoute = Route.Packages, categoryName = "packages" }
-        "Search NixOS packages"
+        [ text "Search more than "
+        , strong [] [ text "80 000 packages" ]
+        ]
         model
         viewSuccess
+        viewBuckets
         SearchMsg
 
 
-viewSuccess :
+viewBuckets :
+    Maybe String
+    -> Search.SearchResult ResultItemSource ResultAggregations
+    -> List (Html Msg)
+viewBuckets bucketsAsString result =
+    let
+        initialBuckets =
+            initBuckets bucketsAsString
+
+        selectedBucket =
+            initialBuckets
+
+        createBucketsMsg getBucket mergeBuckets value =
+            value
+                |> Utils.toggleList (getBucket initialBuckets)
+                |> mergeBuckets initialBuckets
+                |> encodeBuckets
+                |> Json.Encode.encode 0
+                |> Search.BucketsChange
+                |> SearchMsg
+    in
+    []
+        |> viewBucket
+            "Package sets"
+            (result.aggregations.package_attr_set.buckets |> List.sortBy .doc_count |> List.reverse)
+            (createBucketsMsg .packageSets (\s v -> { s | packageSets = v }))
+            selectedBucket.packageSets
+        |> viewBucket
+            "Licenses"
+            (result.aggregations.package_license_set.buckets |> List.sortBy .doc_count |> List.reverse)
+            (createBucketsMsg .licenses (\s v -> { s | licenses = v }))
+            selectedBucket.licenses
+        |> viewBucket
+            "Platforms"
+            (result.aggregations.package_platforms.buckets |> List.sortBy .doc_count |> List.reverse)
+            (createBucketsMsg .platforms (\s v -> { s | platforms = v }))
+            selectedBucket.platforms
+        |> viewBucket
+            "Maintainers"
+            (result.aggregations.package_maintainers_set.buckets |> List.sortBy .doc_count |> List.reverse)
+            (createBucketsMsg .maintainers (\s v -> { s | maintainers = v }))
+            selectedBucket.maintainers
```

Comment:
I would consider simplifying this writter pattern into `List.filterMap` together with chage in api of `viewBucket` function to return maybe.